### ISSUE TITLE
[ENH] fix incomplete deprecation - `Catch22.transform_single_feature`

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#4172]
Deprecation of `Catch22.transform_single_feature` and associated private methods was scheduled for 0.16.0.

While there is no problem with the public method, the private methods cannot be removed, as `_transform_single_feature` is called from `ContinuousIntervalTree`.

This puts us in the situation where logic of `ContinuousIntervalTree` sits in the private intefaces of `Catch22`, which are no longer accessed from `Catch22` itself.

From a design perspective, the private interfaces of the class should not have been accessed in the first place.

Going forward, we need to investigate how to resolve this - e.g., can the same transformation be obtained from `Catch22.transform`.

Good first issue for those confident to do a bit of reverse engineering (unfortunately, the private methods are not too well documented).

**Comment by SandboxHead** [comment]
The private methods in Python need to start with a double underscore rather than a single one. So, these methods are not private. I can see a lot of private methods here with this naming issue.

**Comment by fkiraly** [comment]
@SandboxHead, yes, I was a bit sloppy in terminology, I meant "single-underscore-private" which are called "protected" formally, as I believe.

I don't think we should turn them into "double-underscore-private", design-wise.

**Comment by Aman123lug** [comment]
@fkiraly assign me

**Comment by fkiraly** [comment]
all yours :-)

**Comment by Anamika457** [comment]
@fkiraly Hi, I’d like to work on this issue. Could you please assign it to me?

**Comment by fkiraly** [comment]
@Anamika457 , all yours!
